core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -257,31 +264,54 @@ public AtomicReference<Long> getWorkerId() {
   }
 
   @Override
-  public UfsStatus[] listStatus(String path, ListOptions options) throws IOException {
-    UfsStatus[] statuses = mListStatusCache.getIfPresent(path);
-    if (statuses == null) {
+  public UfsStatus[] listStatus(String path, ListStatusPOptions options) throws IOException {
+    ListOptions listOptions = ListOptions.defaults().setRecursive(
+        options.hasRecursive() ? options.getRecursive() : false);
+
+    long syncIntervalMs = options.hasCommonOptions()
+        ? (options.getCommonOptions().hasSyncIntervalMs()
+        ? options.getCommonOptions().getSyncIntervalMs() : -1) :
+        -1;
+
+    ListStatusResult result = mListStatusCache.getIfPresent(path);
+    if (syncIntervalMs >= 0 && result != null) {
+      // Check if the metadata is still valid.
+      if (System.nanoTime() - result.mTimeStamp > syncIntervalMs * Constants.MS_NANO) {
+        // The metadata is expired. Remove it from in-memory cache.
+        mListStatusCache.invalidate(path);
+        result = null;
+      }
+    }
+
+    if (result == null) {
+      result = new ListStatusResult();
+      result.mTimeStamp = System.nanoTime();
+      result.mUfsStatuses = null;
```

Comment:
Use constructor instead of setting the values afterwards

Reply:
OK

Reply:
Fixed.